hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java

```diff
@@ -37,41 +39,44 @@
  * @param payer payer for the transaction
  * @param status {@link ResponseCodeEnum} status of the transaction
  * @param payerKey payer key required to sign the transaction. It is null if payer is missing
- * @param requiredNonPayerKeys list of keys that are required to sign the transaction, in addition
- *     to payer key
- * @param readKeys the keys that were read during pre-handle
+ * @param payerSignature {@link TransactionSignature} of the payer
+ * @param otherSignatures lit {@link TransactionSignature} of other keys that need to sign
+ * @param innerMetadata {@link TransactionMetadata} of the inner transaction (where appropriate)
  */
 public record TransactionMetadata(
         @Nullable TransactionBody txnBody,
         @Nullable AccountID payer,
+        @Nullable SignatureMap signatureMap,
         @NonNull ResponseCodeEnum status,
         @Nullable HederaKey payerKey,
-        @NonNull List<HederaKey> requiredNonPayerKeys,
-        @Nullable Object handlerMetadata,
-        @NonNull List<ReadKeys> readKeys) {
+        @Nullable TransactionSignature payerSignature,
+        @NonNull Map<HederaKey, TransactionSignature> otherSignatures,
```

Comment:
Are the `HederaKey`'s in this map implicitly restricted to cryptographic keys? (I.e., either Ed25519 or secp256k1 public keys; but not key lists, threshold keys, etc.)

Reply:
These are the `requiredNonPayerKeys` set by the pre-handle methods. In the handle workflow, I need a simple way to find the signature data that belongs to a specific key.